### ISSUE TITLE
NetAdapt: Platform-Aware Neural Network Adaptation for Mobile Applications

### ISSUE THREAD

**Issue by sile** [sile/readings#60]
URL
-----

https://arxiv.org/abs/1804.03230

Abstract
----------

> This work proposes an algorithm, called NetAdapt, that automatically adapts a pre-trained deep neural network to a mobile platform given a resource budget. While many existing algorithms simplify networks based on the number of MACs or weights, optimizing those indirect metrics may not necessarily reduce the direct metrics, such as latency and energy consumption. To solve this problem, NetAdapt incorporates direct metrics into its adaptation algorithm. These direct metrics are evaluated using empirical measurements, so that detailed knowledge of the platform and toolchain is not required. NetAdapt automatically and progressively simplifies a pre-trained network until the resource budget is met while maximizing the accuracy. Experiment results show that NetAdapt achieves better accuracy versus latency trade-offs on both mobile CPU and mobile GPU, compared with the state-of-the-art automated network simplification algorithms. For image classification on the ImageNet dataset, NetAdapt achieves up to a 1.7× speedup in measured inference latency with equal or higher accuracy on MobileNets (V1&V2). 

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
- (3) Methodology: NetAdapt
  - (3-1) Problem Formulation
  - (3-2) Algorithm Overview
  - (3-3) Algorithm Details
  - (3-4) Fast Resource Consumption Estimation
- (4) Experiment Results
  - (4-1) Detailed Settings for MovileNetV1 Experiments
  - (4-2) Comparison with Benchmark Algorithms
  - (4-3) Ablation Studies
  - (4-4) Analysis of Adapted Network Architecture
- (5) Conclusion

BibTeX
------

```bibtex
@inproceedings{yang2018netadapt,
  title={Netadapt: Platform-aware neural network adaptation for mobile applications},
  author={Yang, Tien-Ju and Howard, Andrew and Chen, Bo and Zhang, Xiao and Go, Alec and Sandler, Mark and Sze, Vivienne and Adam, Hartwig},
  booktitle={Proceedings of the European Conference on Computer Vision (ECCV)},
  pages={285--300},
  year={2018}
}
```

**Comment by sile** [comment]
> When executing the algorithm, we look up the table of each layer, and sum up the **layer-wise measurements** to estimate the network-wise resource consumption, which is illustrated in Fig. 3.
